ui/src/singletons/store.ts

```diff
@@ -246,6 +289,11 @@ const useStore = create<AppStore>()(
   )
 )
 
+export const useTemporalStore = <T>(
+  selector: (state: TemporalState<PartializedStoreState>) => T,
+  equality?: (a: T, b: T) => boolean
+) => useStoreWithEqualityFn(useAppStore.temporal, selector, equality)
```

Comment:
I don't think we will need to export the full temporal store as a hook. Seems like only the `undo` and `redo` actions are needed, so maybe we can get away with just:
```
export const useUndoRedo = () =>
  useStore((state) => ({
    undo: state.temporal.undo,
    redo: state.temporal.redo,
  }))
```